libcamera_app.hpp

```diff
@@ -743,8 +744,21 @@ class LibcameraApp
 			}
 			preview_frames_displayed_++;
 			preview_->Show(fd, size, w, h, stride);
+			std::string s = getInfoText(buffer, item.completed_request.metadata);
+			preview_->SetInfoText(s);
 		}
 	}
+	std::string getInfoText(libcamera::FrameBuffer *buffer, libcamera::ControlList &controls)
+	{
+		const libcamera::FrameMetadata &metadata = buffer->metadata();
+		float exposure, analogue_gain, digital_gain, colour_gains[2];
+		char s[128];
+
+		GetFrameInfo(controls, &exposure, &analogue_gain, &digital_gain, colour_gains);
+		snprintf(s, sizeof(s), "Frame #%u: Exp %.2fms AnaGain %.2fx DigGain %.2fx RedGain %.2fx BlueGain %.2fx",
+				 metadata.sequence, exposure/1000.0, analogue_gain, digital_gain, colour_gains[0], colour_gains[1]);
+		return std::string(s);
+	}
```

Comment:
I think it would be nicer to put the sequence number in the CompletedRequest, as we did the framerate. Then it would also be convenient just to pass the CompletedRequest in here as it doesn't need anything else?

The next commit possibly re-arranges some of this stuff, but could this member function be static?

Reply:
Sure. Will fix.